### PR TITLE
Sidekiq does not start with environment variables

### PR DESCRIPTION
**Pull request by johncblandii**
Two different apps need env vars but they aren't passed with the start command. There may be a better way but this one is tested in two stacks and it works.

Code originally done by @codeodor.

### CODE REVIEW

**Comment by codeodor** [comment]
:+1: Thanks for the shoutout and for offering the change back to the public!

**Comment by johncblandii** [comment]
Absolutely hombre! ;)

**Comment by harn** [comment]
+1

File: recipes/setup.rb [hunk 1]
```diff
@@ -63,14 +60,13 @@
...
-        :workers => workers,
-        :syslog => node[:sidekiq][application][:syslog]
-      })
-      notifies :reload, resources(:service => "monit"), :immediately
+      source 'sidekiq_monitrc.erb'
+      variables(deploy: deploy,
+                application: application,
+                workers: workers,
+                syslog: node[:sidekiq][application][:syslog],
+                environment_variables: OpsWorks::Escape.escape_double_quotes(deploy[:environment_variables]))
```
Comment (johncblandii) [inline recipes/setup.rb]: This is really the main change. The rest is just code clean-up by Atom Beautifier.

**Comment by nifuramu** [comment]
+1

**Comment by aadamson** [comment]
+1

File: recipes/configure.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
     mode "0660"
     group deploy[:group]
     owner deploy[:user]
-    variables(:database => deploy[:database], :environment => deploy[:rails_env])
+    variables(:database => deploy[:database], :environment => deploy[:rails_env], :deploy => deploy)
```
Comment (jcavalieri) [inline recipes/configure.rb]: Merge conflict there with PR 2.
https://github.com/drakerlabs/opsworks_sidekiq/pull/2

Comment (jcavalieri) [inline recipes/configure.rb]: Looks like that lines was moved to deploy.rb

Comment (jcavalieri) [inline recipes/configure.rb]: I'm sorry, I misread. That line is gone in PR 2.

Comment (codeodor) [inline recipes/configure.rb]: @johncblandii Not sure how you feel about it, but it seems like that commit may not be relevant to this PR?

Comment (jcavalieri) [inline recipes/configure.rb]: Yeah, sorry for the confusion. I pulled both the open PRs into my local and there is a merge conflict when pulling both in. More of a warning for you all.

**Comment by interpegasus** [comment]
Hi, I would like to know why sidekiq is not starting automatically?

It works manually after running the following command in the opsworks instance.

$ bundle exec sidekiq -e production &

Thanks

**Pull request closed without merging**